### ISSUE TITLE
GHC 8.8 support

### ISSUE THREAD

**Issue by Fuuzetsu** [bgamari/ghc-dump#19]
If I try to build the package with 8.8, I get

```
Building library for ghc-dump-core-0.1.1.0..

on the commandline: warning:
    -Wnoncanonical-monadfail-instances is deprecated: fail is no longer a method of Monad
[1 of 3] Compiling GhcDump.Ast      ( GhcDump/Ast.hs, dist/build/GhcDump/Ast.o )
[2 of 3] Compiling GhcDump.Convert  ( GhcDump/Convert.hs, dist/build/GhcDump/Convert.o )

GhcDump/Convert.hs:196:7: error:
    Not in scope: data constructor ‘Literal.MachChar’
    Perhaps you meant one of these:
      ‘Literal.LitChar’ (imported from Literal),
      ‘Literal.LitChar’ (imported from Literal)
    Module ‘Literal’ does not export ‘MachChar’.
    |
196 |       Literal.MachChar x -> Ast.MachChar x
    |       ^^^^^^^^^^^^^^^^

GhcDump/Convert.hs:197:7: error:
    Not in scope: data constructor ‘Literal.MachStr’
    Module ‘Literal’ does not export ‘MachStr’.
    |
197 |       Literal.MachStr x -> Ast.MachStr x
    |       ^^^^^^^^^^^^^^^

GhcDump/Convert.hs:198:7: error:
    Not in scope: data constructor ‘Literal.MachNullAddr’
    Perhaps you meant one of these:
      ‘Literal.LitNullAddr’ (imported from Literal),
      ‘Literal.LitNullAddr’ (imported from Literal)
    Module ‘Literal’ does not export ‘MachNullAddr’.
    |
198 |       Literal.MachNullAddr -> Ast.MachNullAddr
    |       ^^^^^^^^^^^^^^^^^^^^

GhcDump/Convert.hs:215:7: error:
    Not in scope: data constructor ‘Literal.MachFloat’
    Perhaps you meant one of these:
      ‘Literal.LitFloat’ (imported from Literal),
      ‘Literal.LitFloat’ (imported from Literal)
    Module ‘Literal’ does not export ‘MachFloat’.
    |
215 |       Literal.MachFloat x -> Ast.MachFloat x
    |       ^^^^^^^^^^^^^^^^^

GhcDump/Convert.hs:216:7: error:
    Not in scope: data constructor ‘Literal.MachDouble’
    Perhaps you meant one of these:
      ‘Literal.LitDouble’ (imported from Literal),
      ‘Literal.LitDouble’ (imported from Literal),
      variable ‘Literal.mkLitDouble’ (imported from Literal)
    Module ‘Literal’ does not export ‘MachDouble’.
    |
216 |       Literal.MachDouble x -> Ast.MachDouble x
    |       ^^^^^^^^^^^^^^^^^^

GhcDump/Convert.hs:217:7: error:
    Not in scope: data constructor ‘Literal.MachLabel’
    Perhaps you meant one of these:
      ‘Literal.LitLabel’ (imported from Literal),
      ‘Literal.LitLabel’ (imported from Literal)
    Module ‘Literal’ does not export ‘MachLabel’.
    |
217 |       Literal.MachLabel x _ _ -> Ast.MachLabel $ fastStringToText  x
    |       ^^^^^^^^^^^^^^^^^

GhcDump/Convert.hs:238:25: error:
    Not in scope: data constructor ‘Var.TvBndr’
    Perhaps you meant ‘Var.Bndr’ (imported from Var)
    Module ‘Var’ does not export ‘TvBndr’.
    |
238 | cvtType (Type.ForAllTy (Var.TvBndr b _) t) = Ast.ForAllTy (cvtBinder b) (cvtType t)

```

**Comment by bgamari** [comment]
Mateusz Kowalczyk <notifications@github.com> writes:

> If I try to build the package with 8.8, I get
>
Indeed I haven't yet had time to update this for 8.8. Patches accepted :)

**Comment by maoe** [comment]
FWIW a patch for 8.8 is available at https://github.com/bgamari/ghc-dump/pull/20

**Comment by bgamari** [comment]
Fixed with #21 .
